torch/cuda/_sanitizer.py

```diff
@@ -602,9 +602,20 @@ def enable(self):
         self.dispatch.__enter__()
         self.enabled = True
 
+    def disable(self):
+        self.dispatch.__exit__(None, None, None)
+        self.enabled = False
+
     def __del__(self):
-        if self.enabled:
-            self.dispatch.__exit__(None, None, None)
+        # Since this object lifetime is linked to the `torch.cuda._sanitizer` python
+        # module, it often gets deleted as part of the overal `torch` module cleanup
+        # At that time, depending on CPython version, the torch.* module might be in
+        # different state of already cleaned up.
+        # Similarly other imports might already have been cleaned up so `sys` might
+        # be already gone as well.
+        # Skip exiting the mode if it outlived the runtime.
+        if (sys is not None) and (not sys.is_finalizing) and self.enabled:
```

Comment:
```suggestion
        if (sys is not None) and (not sys.is_finalizing()) and self.enabled:

Reply:
Ho that explains some weird behavior I had with it lol